builder/slide_and_fold_storage.cc

```diff
@@ -107,9 +108,31 @@ class slide_and_fold : public node_mutator {
     int max_workers;
     std::unique_ptr<symbol_map<box_expr>> buffer_bounds;
 
-    loop_info(symbol_id sym, expr orig_min, interval_expr bounds, expr step, int max_workers)
+    var semaphores;
+    var worker_count;
+
+    // How many stages we've added synchronization for.
+    int sync_stages = 0;
+    // If we need synchronization around this stage, which stage it is.
```

Comment:
I was quite confused about stage relates to the loop, but it looks like it just a place the latest stage which was processed, so the stmt visitor can wrap it into semaphores? Is this correct? It really could use more comments.